src/Kentico.Xperience.Siteimprove/Services/SiteimproveService.cs

```diff
@@ -0,0 +1,274 @@
+﻿using System.Net;
+using System.Net.Http.Json;
+
+using CMS.Core;
+using CMS.DataEngine;
+using CMS.Helpers;
+
+using Microsoft.Extensions.Options;
+
+namespace Kentico.Xperience.Siteimprove
+{
+    /// <summary>
+    /// Handles requests to Siteimprove.
+    /// </summary>
+    internal class SiteimproveService : ISiteimproveService
+    {
+        private readonly IHttpClientFactory httpClientFactory;
+        private readonly SiteimproveOptions siteimproveOptions;
+        private readonly IEventLogService eventLogService;
+
+        private long siteID = -1;
+        private string token = string.Empty;
+        private bool isContentCheckEnabled = false;
+
+
+        private HttpClient HttpClient => httpClientFactory.CreateClient(SiteimproveConstants.CLIENT_NAME);

```

Comment:
If you mark this as `private readonly` and then initialize it in the constructor, you won't need to work about creating multiple clients when accessing the `HttpClient` variable.

Usually having a `{ get; }` perform some non-trivial behavior (like creating a new `HttpClient` instance) isn't a good idea because developers are going to assume the value being accessed already exists and might access it multiple times.

By initializing the property in the constructor, you can remove all of the `var client = HttpClient;` lines in your methods and use the `HttpClient` property directly.

Reply:
Changed from property to a `GetHttpClient` method